ewardrobe/ewardrobe_app/views/product.py

```diff
@@ -0,0 +1,17 @@
+from django.views.generic import View
+from ewardrobe_app.models import Product
+from django.shortcuts import render, redirect
+from django.urls import reverse
+
+
+class ProductView(View):
+    template_name = "product.html"
+    failure_url = "main"
+
+    def get(self, request, *args, **kwargs):
+        if request.user.is_authenticated:
+            id = kwargs["id"]
+            product = Product.objects.filter(id=id).first()
```

Comment:
I think `Product.objects.get(id=id)` would work better here, or even:

```suggestion
            product = Product.objects.get(id=kwargs["id"])
```